### PR TITLE
Completed Chitter challenge

### PR DESCRIPTION
**Pull request by nataliehe**
Please kindly review my codes. Any feedback welcome!

### CODE REVIEW

File: lib/database_connection_setup.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+require_relative 'database_connection'
+
+def database_connection_setup
```
Comment (lindseydew) [inline lib/database_connection_setup.rb]: Lovely abstraction here

File: lib/database_connection.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
+require 'pg'
+
+class DatabaseConnection
```
Comment (lindseydew) [inline lib/database_connection.rb]: Lovely wrapper on the DB connection

File: lib/peep.rb [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    peep.gsub!(/\'/, "''")
+    sql = "INSERT INTO peeps (peep, time) VALUES('#{peep}', '#{Time.now}') RETURNING id, peep, time;"
+    result = DatabaseConnection.query(sql)
+    Peep.new(id: result[0]['id'], peep: result[0]['peep'], time: result[0]['time'])
+  end
+
+  def self.all
+    result = DatabaseConnection.query("SELECT * FROM peeps ORDER BY time DESC")
+    result.map do |peep|
+      Peep.new(id: peep['id'], peep: peep['peep'], time: peep['time'])
```
Comment (lindseydew) [inline lib/peep.rb]: Nice!

File: spec/user_spec.rb [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+
+  describe '.authenticate' do
+    it 'returns a user given a correct username and password, if one exists' do
+      user = User.create(name: 'John', email: 'test@example.com', password: 'password123')
+      authenticated_user = User.authenticate(email: 'test@example.com', password: 'password123')
+      expect(authenticated_user.id).to eq user.id
+    end
+
+    it 'returns nil given an incorrect email address' do
+      User.create(name: 'John', email: 'test@example.com', password: 'password123')
```
Comment (lindseydew) [inline spec/user_spec.rb]: Nice testing of edge cases here

File: spec/spec_helper.rb [hunk 5]
```diff
@@ -15,3 +20,15 @@
...
+
+ENV['RACK_ENV'] = 'test'
+
+Capybara.app = Chitter
+
+Rake.application.load_rakefile
+
+RSpec.configure do |config|
+  config.before(:each) do
+    Rake::Task['clear_test_database'].execute
```
Comment (lindseydew) [inline spec/spec_helper.rb]: Nice!

File: spec/peep_spec.rb [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+
+describe Peep do
+  describe '.create' do
+    it 'creates a new peep' do
+      msg = 'Coding is fun!'
+      peep = Peep.create(peep: msg)
+      data = data(table: 'peeps', id: peep.id)
+      expect(peep.id).to eq(data['id'])
+      expect(peep.peep).to eq(msg)
+      expect(peep.time).to match(/(\d{4})-(\d{2})-(\d{2}) (\d{2}):(\d{2}):(\d{2})/)
```
Comment (lindseydew) [inline spec/peep_spec.rb]: You could make use of a mock here for the time so that you can assert that `peep.time` gives a timestamp.
